From 9e2c54969f9e3114875a55d62cddee3ef76ba4a0 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 4 Jul 2024 18:51:22 +0200
Subject: [PATCH 1169/1695] dts: stm32: stm32mp2* wakeup GPIO are secure

Define in stm32mp2* boards DTS files that PWR wakeup pins relates
to GPIOs assigned to OP-TEE (Cortex-A35 privileged secure world).

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1a15a568e4838f3f4871b9403d732e94675dac0d dts: stm32: stm32mp2* wakeup GPIO are secure]
Change-Id: I17aa13f452684109bfe64de8d6bbc592070dcff7
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/391591
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rif.dtsi  | 2 +-
 core/arch/arm/dts/stm32mp235f-dk-ca35tdcid-rif.dtsi  | 2 +-
 core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi  | 2 +-
 core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rif.dtsi b/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rif.dtsi
index 9945aeb18..212a44e19 100644
--- a/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rif.dtsi
+++ b/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rif.dtsi
@@ -274,7 +274,7 @@
 
 &gpioa {
 	st,protreg = <
-		RIFPROT(RIF_IOPORT_PIN(0), RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
+		RIFPROT(RIF_IOPORT_PIN(0), RIF_UNUSED, RIF_UNLOCK, RIF_SEC, RIF_PRIV, RIF_CID1, RIF_SEM_DIS, RIF_CFEN)
 		RIFPROT(RIF_IOPORT_PIN(1), RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(RIF_IOPORT_PIN(2), RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(RIF_IOPORT_PIN(3), RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
diff --git a/core/arch/arm/dts/stm32mp235f-dk-ca35tdcid-rif.dtsi b/core/arch/arm/dts/stm32mp235f-dk-ca35tdcid-rif.dtsi
index ee003bf02..bbad4f213 100644
--- a/core/arch/arm/dts/stm32mp235f-dk-ca35tdcid-rif.dtsi
+++ b/core/arch/arm/dts/stm32mp235f-dk-ca35tdcid-rif.dtsi
@@ -335,7 +335,7 @@
 
 &gpioa {
 	st,protreg = <
-		RIFPROT(RIF_IOPORT_PIN(0), RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
+		RIFPROT(RIF_IOPORT_PIN(0), RIF_UNUSED, RIF_UNLOCK, RIF_SEC, RIF_PRIV, RIF_CID1, RIF_SEM_DIS, RIF_CFEN)
 		RIFPROT(RIF_IOPORT_PIN(1), RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(RIF_IOPORT_PIN(2), RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(RIF_IOPORT_PIN(3), RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
diff --git a/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi b/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi
index 34f5dea47..68640aa82 100644
--- a/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi
+++ b/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi
@@ -328,7 +328,7 @@
 
 &gpioa {
 	st,protreg = <
-		RIFPROT(RIF_IOPORT_PIN(0), RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
+		RIFPROT(RIF_IOPORT_PIN(0), RIF_UNUSED, RIF_UNLOCK, RIF_SEC, RIF_PRIV, RIF_CID1, RIF_SEM_DIS, RIF_CFEN)
 		RIFPROT(RIF_IOPORT_PIN(1), RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(RIF_IOPORT_PIN(2), RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(RIF_IOPORT_PIN(3), RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
diff --git a/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi b/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi
index b63afb94f..7a2019c00 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi
+++ b/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi
@@ -328,7 +328,7 @@
 
 &gpioa {
 	st,protreg = <
-		RIFPROT(RIF_IOPORT_PIN(0), RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
+		RIFPROT(RIF_IOPORT_PIN(0), RIF_UNUSED, RIF_UNLOCK, RIF_SEC, RIF_PRIV, RIF_CID1, RIF_SEM_DIS, RIF_CFEN)
 		RIFPROT(RIF_IOPORT_PIN(1), RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(RIF_IOPORT_PIN(2), RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(RIF_IOPORT_PIN(3), RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
-- 
2.47.2

